### PR TITLE
Streamline export application resource loading

### PR DESCRIPTION
**Pull request by jgustie**
This PR eliminates a couple of extra round trips we are doing during export as part of the application refactoring.

### CODE REVIEW

File: internal/scan/runtime.go [hunk 1]
```diff
@@ -103,10 +103,10 @@ func (o *ObjectList) Write(nodes []*yaml.RNode) error {
 
 		obj := newObjectForKind(u)
 		if err := Scheme.Convert(u, obj, runtime.InternalGroupVersioner); err != nil {
-			return err
+			o.Items = append(o.Items, runtime.RawExtension{Raw: data})
```
Comment (jgustie) [inline internal/scan/runtime.go:106]: Previously there was a comment about CRDs, technically this is currently only being used by resources that are part of experiment generation so it isn't a problem. But if we were to `IncludeApplicationResources` on the generator this change would ensure that custom resources would effectively pass through without conversion.

File: internal/scan/runtime.go [hunk 2]
```diff
@@ -144,6 +144,7 @@ func fixConversion(node *yaml.RNode) (*yaml.RNode, error) {
 func newObjectForKind(obj runtime.Object) runtime.Object {
 	if gvks, _, _ := Scheme.ObjectKinds(obj); len(gvks) > 0 {
 		if typed, err := Scheme.New(gvks[0]); err == nil {
+			typed.GetObjectKind().SetGroupVersionKind(gvks[0])
```
Comment (jgustie) [inline internal/scan/runtime.go:147]: Export was doing an extra Unstructured conversion to get this information, it's easier to just add while we have it.

File: redskyctl/internal/commands/export/export.go [hunk 3]
```diff
@@ -422,17 +402,19 @@ func (o *Options) generateExperiment() error {
 	}
 
 	// Load up all application resources
-	res, err := apppkg.LoadResources(o.application, onDiskFS)
-	if err != nil {
-		return err
-	}
-
-	resYaml, err := res.AsYaml()
```
Comment (jgustie) [inline redskyctl/internal/commands/export/export.go:430]: We can just use a KYAML pipeline and skip the extra round trip into a Kustomize `ResMap`.

Comment (bradbeam) [reply redskyctl/internal/commands/export/export.go:430]: I think kio pipelines have taken over your mind :)

**Review by bradbeam — COMMENTED** [review]
I think this looks good, unsure what's going on with CI

File: redskyctl/internal/commands/export/export.go (hunk 3 above)
Comment (jgustie) [reply redskyctl/internal/commands/export/export.go:430]: 100%

**Pull request merged**